[Finder] Command::addAtIndex() fails with Command instance argument

thunderer opened the pull request:
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #14384 |
| License | MIT |
| Doc PR | - |

Fixed reported bug #14384 and added test case for it.

Commit:
fixes issue with logging array of non-utf8 data

lancechentw committed:
Handled bearer authorization header in REDIRECT_ form

Apache rewrite module renames client request
header (`HTTP_`) by prepending `REDIRECT_` to
it. http basic authentication and http digest
authentication are properly processed in
REDIRECT_ form, while bearer is processed in
HTTP_ form, but dropped in REDIRECT_ form.

nicovak committed:
WebProfiler break words

WebProfiler CSS word-break: break-all;

BruceWouaigne committed:
[Form] [EventListener] fixed sending non array data on submit to ResizeListener

cordoval committed:
[DX] improve file loader error for router/other resources in bundle

gharlan committed:
CS: Pre incrementation/decrementation should be used if possible

stefanruijsenaars committed:
Fixed compatibility with PHP7 and up by introducing new constraints (IsNull, IsTrue, IsFalse) and related validators (IsNullValidator, IsTrueValidator, IsFalseValidator)

aitboudad committed:
[TwigBundle] Refresh twig paths when resources change.

hason committed:
[HttpKernel] Check if "symfony/proxy-manager-bridge" package is installed

nicolas-grekas committed:
[FrameworkBundle] Workaround php -S ignoring auto_prepend_file

xabbuh committed:
improved exception when missing required component

danez committed:
[HttpFoundation] Fix baseUrl when script filename is contained in pathInfo

dupuchba committed:
Add better phpdoc message for getListeners method of the EventDispatcher

xabbuh committed:
[EventDispatcher] make listeners removable from an executed listener

nicolas-grekas committed:
[FrameworkBundle] Fix Routing\DelegatingLoader resiliency to fatal errors

keradus committed:
CS fixes

keradus committed:
CS fixes

nicolas-grekas committed:
[HttpKernel] Cleanup ExceptionListener

thunderer committed:
[Finder] fixed bug #14384: Command::addAtIndex() fails with PHP error when Command instance is passed

xabbuh commented on src/Symfony/Component/Finder/Tests/FinderTest.php:
```diff
@@ -852,4 +853,15 @@ public function testMultipleLocationsWithSubDirectories()
+    public function testShellCommand()
+    {
+        $cmd = Command::create()->add('--force');
+        $this->assertSame('--force', $cmd->join());
```
This is not related to the issue, is it?

thunderer replied:
It's just a sanity check before checking problem case. `Command` class is not tested anywhere directly, so I thought it would be a good idea.

xabbuh replied:
Maybe it's a better idea to split this into more than one test then.

By the way, it could be interesting to use index `1` instead of `2` in the second call of `addAtIndex()` to ensure that you can also insert in the middle.

thunderer replied:
How about introducing separate `Shell\CommandTest.php` suite to cover `Command` class then? I'd be happy to work on it as it's one of the least covered class in a whole component.

thunderer replied:
@xabbuh Can you reply to my comment? I'd be happy to refactor the tests, but I need guidance on what effect are you interested in.

xabbuh replied:
@thunderer A dedicated `CommandTest` class sounds like a good idea to me.

xelaris committed:
[FrameworkBundle] Check for 'xlf' instead of 'xliff'

There is no translation writer format named 'xliff', but 'xlf' only. So the TranslationUpdateCommand can't be called with 'output-format' == 'xliff' and the version info will never be shown.

aitboudad committed:
[Framework] added test for Router commands.

keradus committed:
CS: unalign =>

dosten committed:
Show a better error when the port is in use

keradus committed:
CS: unalign =

Restless-ET committed:
Fix Portuguese (Portugal) translation for Security

pmodin committed:
[StringUtil] Fixed singularification of 'selfies'

dosten committed:
Fixed tests

webmozart committed:
[Validator] Fixed Choice when an empty array is used in the "choices" option

dosten committed:
[DependencyInjection] Removed extra strtolower calls

Commit:
Use https://symfony.com/search for searching

Using https for symfony.com/search stops chrome (and eventually firefox)
from warning us about "Mixed Content" when developing sites that use SSL
for the entire site.

thunderer committed:
removed sample Command test from FinderTest, added Command test suite

thunderer commented:
@xabbuh I've added tests for `Shell\Command` class, 100% coverage from now. PHP 5.3.3 build errored due to network problem with Composer. Please tell me what you think.

BTW Since I'm the author of the `Tests\Shell\CommandTest.php` file, can I add @author docblock above `class` declaration?

dosten commented on src/Symfony/Component/Finder/Tests/Shell/CommandTest.php:
```diff
@@ -0,0 +1,162 @@
+        $cmd->add('--force');
+        $cmd->add('--run');
+
+        $this->assertSame('--force --run', (string)$cmd);
```
missing space: `(string) $cmd`

thunderer replied:
Thanks, I'll fix this in one batch when @xabbuh gets back to me with his thoughts. :)

xabbuh replied:
Sorry, I missed this. Yes, please change as suggested. :)

aitboudad committed:
[profiler][security] check authenticated user by tokenClass instead of username.

sgrodzicki committed:
[HttpFoundation] Allow curly braces in trusted host patterns

lyrixx committed:
[FrameworkBundle] Added missing log in server:run command

Because of the current implementation of `php -S` command,
logs are skipped if a front controller is specified.

This patch put back the missing logs

xabbuh committed:
link to https://symfony.com where possible

xphere committed:
add more entropy to generated classnames

- use proxy destination filename to create and md5 salt
- apply fabbot patch
- inline salt generation

wouterj committed:
Use HTTPS in README and some other fixes

dosten committed:
Run tests on hhvm instead of hhvm-nightly

nicolas-grekas committed:
[Debug 2.3] Fix test for PHP7

kix committed:
[Filesystem] Simplified an if statement

Commit:
[SecurityBundle] Use Enum Nodes Instead Of Scalar

MacDada committed:
PhpDoc fix in AbstractRememberMeServices

All extending classes return `UserInterface`, not TokenInterface:

* https://github.com/symfony/symfony/blob/2.3/src/Symfony/Component/Security/Http/RememberMe/TokenBasedRememberMeServices.php#L64
* https://github.com/symfony/symfony/blob/2.3/src/Symfony/Component/Security/Http/RememberMe/PersistentTokenBasedRememberMeServices.php#L116

And `AbstractRememberMeServices` actually required the return value to be `UserInterface`:

            $user = $this->processAutoLoginCookie($cookieParts, $request);

            if (!$user instanceof UserInterface) {
                throw new \RuntimeException('processAutoLoginCookie() must return a UserInterface implementation.');
            }

Tobion committed:
[SecurityBundle] use session auth constants in config

asiragusa committed:
Avoid redirection to XHR URIs

fabpot committed:
updated CHANGELOG for 2.3.28

fabpot committed:
update CONTRIBUTORS for 2.3.28

fabpot committed:
updated VERSION for 2.3.28

MatTheCat committed:
[Security][Translation] fixes #14584

fabpot committed:
bumped Symfony version to 2.3.29

TomasVotruba committed:
ContainerInterface: unused exception dropped

dosten committed:
Fix tests in HHVM

nicolas-grekas committed:
[travis] Use container-based infrastructure

jakzal committed:
[DomCrawler] Throw an exception if a form field path is incomplete.

nicolas-grekas committed:
[travis] Don't use the cache

nicolas-grekas committed:
fix missing links to https://symfony.com

nicolas-grekas committed:
[HttpFoundation] Fix volatile MongoDbSessionHandlerTest::testRead()

jakzal committed:
[HttpKernel] Handle an array vary header in the http cache store

nicolas-grekas committed:
[DependencyInjection] Avoid unnecessary calls to strtolower()

fabpot committed:
fixed CS

94noni committed:
Update README.md

fabpot committed:
fixed typo

MacDada committed:
[Security] TokenBasedRememberMeServices test to show why encoding username is required

MacDada committed:
[Security] AbstractRememberMeServices::encodeCookie() validates cookie parts

King2500 committed:
[FrameworkBundle] Removed unnecessary parameter in TemplateController

Response::setPublic doesn't have any parameters, so this parameter call is not needed.

nanocom committed:
[Console] Delete duplicate test in CommandTest

There is no __get method in the Command class, and the deleted test was duplicated with the preceding one.

zerkms committed:
[HttpFoundation] IpUtils::checkIp4() should allow  networks

nicolas-grekas committed:
Fix HTML escaping of to-source links

aitboudad committed:
[console][formater] allow format toString object.

jakzal committed:
[HttpKernel] Do not call the FragmentListener if _controller is already defined

dosten committed:
Check instance of FormBuilderInterface instead of FormBuilder

SpacePossum committed:
Fix typo

taueres committed:
[HttpKernel][Bundle] Check extension implements ExtensionInterface

- Avoid fatal errors on line 89 (calling getAlias on objects of
  unknown type).
- Help developers solve problems with their extensions

dunglas committed:
[HttpFoundation] Get response content as resource several times for PHP >= 5.6

ogizanagi committed:
[CS] [Console] StreamOuput : fix loose comparison

dosten committed:
Improved duplicated code in FileLocator

fabpot committed:
fixed C

amouhzi committed:
[Validators] Missing translations for arabic language.

fabpot committed:
updated CHANGELOG for 2.3.29

fabpot committed:
update CONTRIBUTORS for 2.3.29

fabpot committed:
updated VERSION for 2.3.29

fabpot committed:
bumped Symfony version to 2.3.30

fabpot committed:
added missing CVE number

thunderer commented:
@xabbuh Can you comment on the current state of this PR?

DancZer committed:
InvalidResourceException file name

amouhzi committed:
[Validators] Correct translation key and content [nl]

amouhzi committed:
[Validators] Remove forgotten space in a translation key [nl]

fabpot committed:
Revert "bug #14262 [TwigBundle] Refresh twig paths when resources change. (aitboudad)"

This reverts commit 4d408525962b78abbb9f40d699d1b959aae2f8f1, reversing
changes made to dd2fb850a7795c1092c8f6efe1db959f09b4ac02.

kalessil committed:
Php Inspections (EA Extended):
    - resolved possible PHP Fatal in \Symfony\Component\BrowserKit\Cookie::__toString
    -resolved implicit magic methods calls
    -resolved callable name case mismatches

fabpot committed:
updated CHANGELOG for 2.3.30

amouhzi committed:
[BrowserKit] Fix bug when uri starts with http.

| Q             | A
| ------------- | ---
| Fixed tickets |
| License       | MIT

fabpot committed:
updated VERSION for 2.3.30

fabpot committed:
bumped Symfony version to 2.3.31

xabbuh committed:
[Security][Acl] enforce string identifiers

stof committed:
Improve the config validation in TwigBundle

kalessil committed:
[2.3] Static Code Analysis for Components

c960657 committed:
[Form] Support DateTimeImmutable in transform()

taueres committed:
[Form] Fix call to removed method (BC broken in 2.3)

ResolvedFormType#getTypeExtensions() was calling method
AbstractType#getExtensions() which has been removed in
commit b3081e85a062abb8fc8a22f20a2b3289db2ff501

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

azine committed:
Allow new lines in Messages translated with transchoice() (replacement for #14867)

wouterj committed:
[2.3] Fix tests on Windows

rvanlaak committed:
[Security] Initialize SwitchUserEvent::targetUser on attemptExitUser

The `SwitchUserEvent` is triggered in case an account is switched. This works okay while switching to the user, but on exit the `SwitchUserEvent` is triggered again with the original User. That User was not initialized by the provider yet.

load user by UserInterface instead of username

javiereguiluz committed:
Standardize the name of the exception variables

fabpot committed:
fixed CS

webmozart committed:
[Form] Fixed: Filter non-integers when selecting entities by int ID

fabpot commented:
:+1: 

Can you rebase and fix CS issues?

xabbuh commented on src/Symfony/Component/Finder/Tests/FinderTest.php:
```diff
@@ -13,6 +13,7 @@
 
 use Symfony\Component\Finder\Finder;
 use Symfony\Component\Finder\Adapter;
+use Symfony\Component\Finder\Shell\Command;
```
This should be reverted.

MacDada committed:
[bugfix][MonologBridge] WebProcessor: passing  to BaseWebProcessor

dosten committed:
Remove duplicated paths

Restless-ET committed:
[Translation][Form][choice] empty_value shouldn't be translated when it has an empty value

nicolas-grekas committed:
[2.3][Debug] Fix fatal-errors handling on HHVM

nicolas-grekas committed:
[DoctrineBridge] Bypass the db when no valid identifier is provided in ORMQueryBuilderLoader

aitboudad committed:
[Translation][update cmd] taken account into bundle overrides path.

kalessil committed:
[2.3] Static Code Analysis for Components

johnkary committed:
[Console] Fix STDERR output text on IBM iSeries OS400

webmozart committed:
[Form] Fixed: remove quoted strings from Intl date formats (e.g. es_ES full pattern)

webmozart committed:
[Form] Fixed: Data mappers always receive forms indexed by their names

nicolas-grekas committed:
[HttpFoundation] Use convention to allow throwing from __toString()

nicolas-grekas committed:
[DependencyInjection] Fail when dumping a Definition with no class nor factory

javiereguiluz committed:
Fixed the regexp for the validator of Maestro-based credit/debit cards

garak committed:
Fix typo in Italian translation

dawehner committed:
Add a way to reset the singleton

xabbuh committed:
fix validation for Maestro UK card numbers

weaverryan committed:
No change - the normalizeParams is a copy-and-paste of the earlier logic

weaverryan committed:
Normalizing recursively - see #9096

ogizanagi committed:
Fix quoting style consistency.

xabbuh committed:
trigger event with right user (add test)

nicolas-grekas committed:
Towards 100% HHVM compat

shehi committed:
Azerbaijani language pluralization rule

In AZ, as in TR, pluralization is always 0: 

0 kitab (zero books)
1 kitab (1 book)
3 kitab (3 books)
104 kitab (104 books)

Apparently ZF ruleset was wrong in the first place :)

nicolas-grekas committed:
Minor fixes

nicolas-grekas committed:
[Stopwatch] Fix test

zhil committed:
Update DateTimeToArrayTransformer.php

nicolas-grekas committed:
Move HHVM tests out of the allowed failures

nicolas-grekas committed:
Mock microtime() and time() in transient tests

nicolas-grekas committed:
[travis] start hhvm first

nicolas-grekas committed:
Enhance hhvm test skip message

Commit:
Remove var not used due to returning early (introduced in 8982c32)

fabpot committed:
fixed CS

keradus committed:
fix CS

jakzal commented:
ping @thunderer

thunderer committed:
fixed CS issues

thunderer commented:
@jakzal I fixed CS issues and rebased my branch for 2.3 but it added the whole history to my branch. Can you merge it like that or is it bad and I should do something about that?

jakzal commented:
@thunderer it might be hard. Are you sure you rebased with the latest 2.3?

thunderer commented:
I don't know what happened, I just fetched upstream/2.3, pulled it, then rebased my branch and strange things happened. Please see PR #15223, hopefully it'll be okay.
